config_generator/src/main.rs

```diff
@@ -0,0 +1,191 @@
+    let sxg_config: SxgConfig = serde_yaml::from_str(&wrangler_config.vars.sxg_config).unwrap();
+    (wrangler_config, sxg_config)
+}
+
+fn main() -> Result<(), std::io::Error> {
+    goto_repository_root()?;
+    let (mut wrangler_config, mut sxg_config) = read_exsiting_config();
+    loop {
+        let cert = read_certificate_pem_file("credentials/cert.pem");
+        let issuer = read_certificate_pem_file("credentials/issuer.pem");
+        if cert.is_ok() && issuer.is_ok() {
+            println!("Succesffuly read certificates");
+            wrangler_config.vars.cert_pem = cert.unwrap();
+            wrangler_config.vars.issuer_pem = issuer.unwrap();
+            break;
+        }
+        if let Err(msg) = cert {
+            println!("{}", msg);
+        }
+        if let Err(msg) = issuer {
+            println!("{}", msg);
+        }
+        println!(
+            "To generate certificates, please refer to\n\
+            \thttps://github.com/google/sxg-rs/blob/main/credentials/README.md"
+        );
+        Input::<String>::new()
+            .with_prompt("Press ENTER key after you generate the certificates")
+            .allow_empty(true)
+            .interact_text()
+            .unwrap();
+    }
+    let user = get_global_uesr();
+    wrangler_config.account_id = Input::new()
+        .with_prompt("What's your Cloudflare account ID?")
+        .with_initial_text(wrangler_config.account_id)
+        .interact_text()
+        .unwrap();
+    wrangler_config.zone_id = Input::new()
+        .with_prompt("What's your Cloudflare zone ID?")
+        .with_initial_text(wrangler_config.zone_id)
+        .interact_text()
+        .unwrap();
+    sxg_config.html_host = Input::new()
+        .with_prompt("What's your domain registered in Cloudflare?")
+        .with_initial_text(sxg_config.html_host)
+        .validate_with(|s: &String| -> Result<(), url::ParseError> {
+            url::Host::parse(s)?;
+            Ok(())
+        })
+        .interact_text()
+        .unwrap();
+    sxg_config.worker_host = Input::new()
+        .with_prompt("What's the domain of your Cloudflare worker?")
+        .with_initial_text(sxg_config.worker_host)
+        .validate_with(|s: &String| -> Result<(), url::ParseError> {
+            url::Host::parse(s)?;
+            Ok(())
+        })
+        .interact_text()
+        .unwrap();
+    // TODO: Allow user to customize routes
+    wrangler_config.routes = vec![format!("{}/*", sxg_config.html_host)];
+    let ocsp_kv_id = get_ocsp_kv_id(&user, &wrangler_config.account_id);
+    wrangler_config.kv_namespaces = vec![ConfigKvNamespace {
+        binding: String::from("OCSP"),
+        id: Some(ocsp_kv_id),
+        preview_id: None,
+    }];
+    wrangler_config.vars.sxg_config = serde_yaml::to_string(&sxg_config).unwrap();
+    std::fs::write(
+        CONFIG_FILE,
+        toml::to_string_pretty(&wrangler_config).unwrap(),
```

Comment:
If possible, add a comment at the top that this file is generated and comments won't be preserved.

Reply:
Done